internal/internal.go

```diff
@@ -147,6 +152,8 @@ type envoyExtAuthzGrpcServer struct {
 	interQueryBuiltinCache iCache.InterQueryCache
 }
 
+var _ envoyauth.EvalContextWithTarget = (*envoyExtAuthzGrpcServer)(nil)
```

Comment:
Without having any other effects, it asserts that `*envoyExtAuthzGrpcServer` fulfills the `envoyauth.EvalContextWithTarget` interface.

Usually, you wouldn't need that, if there was some method like
```go
func NewThing() envoyauth.EvalContextWithTarget {
    return &envoyExtAuthzGrpcServer{...}
```
then the compiler already ensures that `*envoyExtAuthzGrpcServer` satisfies the interface.

In _this case_, however, we're adding something in an unobtrusive way, so we better assert this ourselves. Imagine that some time in the future, we'd change the method name from `Target()` to `Schmarget()` (because why not) -- without this assertion here, the code would still compile just fine if we didn't rename the method defined on the `envoyExtAuthzGrpcServer` struct. With this assertion, we'll be reminded to do that.

Reply:
Thanks for the clarification !